infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -97,12 +98,17 @@ public CorfuServerNode(@Nonnull ServerContext serverContext,
      * Start the Corfu Server by listening on the specified port.
      */
     public ChannelFuture start() {
+        String host = serverContext.getConfiguration().getHostAddress();
+        String interfaceToBind = serverContext.getConfiguration().getNetworkInterface();
+        if (interfaceToBind != null) {
+            host = NetworkUtils.getAddressFromInterfaceName(interfaceToBind);
+        }
```

Comment:
Maybe this can be moved to `ServerConfiguration` class when built? Or is there a need for these to remain separate? I noticed that this check is not performed in `NettyLogReplicationServerChannelAdapter::startServer`.

Reply:
The check is already performed in `ServerConfiguration`:
`String networkInterfaceName = configProperties.getProperty(NETWORK_INTERFACE);
        String address = configProperties.getProperty(HOST_ADDRESS);
        if (networkInterfaceName != null) {
            conf.setHostAddress(getAddressFromInterfaceName(networkInterfaceName));
            conf.setNetworkInterface(networkInterfaceName);
            conf.setBindToAllInterfaces(false);
        } else if (address == null) {
            // Default the address to localhost and set the bind to all interfaces flag to true,
            // if the address and interface is not specified.
            conf.setBindToAllInterfaces(true);
            conf.setHostAddress("localhost");
        } else {
            // Address is specified by the user.
            conf.setHostAddress(address);
            conf.setBindToAllInterfaces(false);
        }`
Removing redundant check.